### ISSUE TITLE
OPM fails to create index with Podman 3.0.1

### ISSUE THREAD

**Issue by dubee** [operator-framework/operator-registry#689]
There was a change introduced in Podman after version` 2.2.1` to display a warning message when the owner of the current cgroup could not be detected. When using OPM `v1.14.3-34` with Podman `3.0.1-7` to add a bundle to an index, OPM fails because it does not expect this warning message. I have traced this problem to runner.go where OPM tries to run `podman create`. When trying to extract the ID from the create output, OPM fails as it does not expect the warning message to be in the output (`id := strings.TrimSuffix(string(out), "\n")`.

I am running rootless in a container with this base image: `registry.redhat.io/rhel8/podman:8.4`. Runnnig `yum update -y` in the Dockerfile gives me Podman `3.0.1-7`.

Here is the line of code where the problem is:
https://github.com/operator-framework/operator-registry/blob/master/pkg/containertools/runner.go#L139. Though the warning may cause problems with other Podman commands that OPM runs.

OPM failure log:
```
...
INFO[0003] running podman cp"
ERRO[0003] time="2021-06-23T09:27:21-04:00" level=warning msg="Failed to detect the owner for the current cgroup: stat /sys/fs/cgroup/systemd/docker/350ea3d90cc8a53bae50e9d52e0a2ba7fb1320b02b5649d9299e032481b38a1d: no such file or directory"
Error: container "time=\"2021-06-23T09" does not exist"
ERRO[0003] permissive mode disabled                      error="error copying container directory time=\"2021-06-23T09:27:21-04:00\" level=warning msg=\"Failed to detect the owner for the current cgroup: stat /sys/fs/cgroup/systemd/docker/350ea3d90cc8a53bae50e9d52e0a2ba7fb1320b02b5649d9299e032481b38a1d: no such file or directory\"\nError: container \"time=\\\"2021-06-23T09\" does not exist\n: exit status 125"
Error: error copying container directory time="2021-06-23T09:27:21-04:00" level=warning msg="Failed to detect the owner for the current cgroup: stat /sys/fs/cgroup/systemd/docker/350ea3d90cc8a53bae50e9d52e0a2ba7fb1320b02b5649d9299e032481b38a1d: no such file or directory"
Error: container "time=\"2021-06-23T09" does not exist
```

**Comment by dubee** [comment]
@njhale, can you help with this?

**Comment by exdx** [comment]
@dubee is this still an issue in the newer version of opm? I believe it is, unless podman has updated something on its end. 

If it's still an issue, we can work on addressing it. We are currently working on changing some of the unpacking code and can looking into this issue as well.

**Comment by exdx** [comment]
Seems like we are parsing both stdout and stderr when looking for the containerid, perhaps if we changed it to only look at stdout this warning message would not interfere

See https://github.com/operator-framework/operator-registry/blob/master/pkg/containertools/runner.go#L133

**Comment by dubee** [comment]
@exdx, thanks for your quick response! 

I'm using the latest OPM. Changing the runner.go code to not combined stdout and stderr does seems to fix the problem.

**Comment by dubee** [comment]
@ankitathomas , any updates on this one or a time frame for when it will be resolved? We are required to update to the latest version of Podman, but cannot due to this problem with OPM.

**Comment by joelanford** [comment]
@dubee Is there a PR with the changes to fix this? If not, any chance you can submit one with your fixes to `runner.go`?

**Comment by dinhxuanvu** [comment]
We need to review the pending PR and after it is merged, we can close this issue.

**Comment by ankitathomas** [comment]
The specific error causing this has been fixed in podman https://github.com/containers/podman/pull/9858, available as of 3.1.2

The linked PR can still be merged to avoid mixing stdout and stderr messages for podman though, in case of future stderr warnings
